### PR TITLE
boards/dwm1001: add support to Decawave DWM1001 dev board

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR adds support for the [Decawave DWM1001 dev board](https://www.decawave.com/product/dwm1001-development-board/). This boards runs on a nrf52832 microcontroller and also provides a UWB tranceiver and LIS2DH12 motion sensor.

What has been tested so far:
- UART
- I2C (with lis2dh12)
- compile_and_test_for_board.py is running, just waiting for it to complete but for the moment, things are going well.

### Testing procedure

- A green Murdock
- lis2dh12 is initialized and can queried via `examples/saul`
- I'll post there the results of `compile_and_test_for_board.py` when they are done

### Issues/PRs references

None

### CODE REVIEW

**Comment by benpicco** [comment]
I'd call it `dwm1001-dev` to avoid confusion with the radio driver.

File: boards/dwm1001/include/periph_conf.h [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ */
+static const spi_conf_t spi_config[] = {
+    {
+        .dev  = NRF_SPIM0,
+        .sclk = GPIO_PIN(0, 4),
+        .mosi = GPIO_PIN(0, 5),
+        .miso = GPIO_PIN(0, 6),
+#ifdef ERRATA_SPI_SINGLE_BYTE_WORKAROUND
+        .ppi = 0,
+#endif
```
Comment (benpicco) [inline boards/dwm1001/include/periph_conf.h:53]: You could just drop that - it will be 0 either way.

File: boards/dwm1001/include/periph_conf.h [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+/**
+ * @name    I2C configuration
+ * @{
+ */
+static const i2c_conf_t i2c_config[] = {
+    {
+        .dev = NRF_TWIM1,
+        .scl = 28,
+        .sda = 29,
```
Comment (benpicco) [inline boards/dwm1001/include/periph_conf.h:77]:
```suggestion
        .scl = GPIO_PIN(0, 28),
        .sda = GPIO_PIN(0, 29),
```

result will be the same, but looks more consistent

**Comment by aabadie** [comment]
> I'd call it dwm1001-dev to avoid confusion with the radio driver.

The radio driver should be called dw1001 so there won't be a name clash.

File: boards/dwm1001/include/periph_conf.h (hunk 2 above)
Comment (aabadie) [reply boards/dwm1001/include/periph_conf.h:77]: Already done locally, just forgot to push.

**Comment by benpicco** [comment]
Feel free to squash already

**Comment by aabadie** [comment]
Here are the results of the `compile_and_test_for_board.py` script:

```
Failures during test:
- [examples/micropython](examples/micropython/test.failed)
- [examples/suit_update](examples/suit_update/test.failed)
- [tests/emcute](tests/emcute/test.failed)
- [tests/gnrc_dhcpv6_client](tests/gnrc_dhcpv6_client/test.failed)
- [tests/gnrc_dhcpv6_client_6lbr](tests/gnrc_dhcpv6_client_6lbr/test.failed)
- [tests/gnrc_ipv6_ext](tests/gnrc_ipv6_ext/test.failed)
- [tests/gnrc_ipv6_ext_frag](tests/gnrc_ipv6_ext_frag/test.failed)
- [tests/gnrc_ipv6_ext_opt](tests/gnrc_ipv6_ext_opt/test.failed)
- [tests/gnrc_ipv6_nib_dns](tests/gnrc_ipv6_nib_dns/test.failed)
- [tests/gnrc_rpl_srh](tests/gnrc_rpl_srh/test.failed)
- [tests/gnrc_sock_dns](tests/gnrc_sock_dns/test.failed)
- [tests/pkg_libfixmath_unittests](tests/pkg_libfixmath_unittests/test.failed)
- [tests/xtimer_now_irq](tests/xtimer_now_irq/test.failed)

Failures during test.flash:
- [tests/gnrc_tcp](tests/gnrc_tcp/test.flash.failed)
- [tests/nordic_softdevice](tests/nordic_softdevice/test.flash.failed)
- [tests/periph_timer_short_relative_set](tests/periph_timer_short_relative_set/test.flash.failed)
```

The flash failure with `tests/nordic_softdevice` seems critical but I'm not sure it's very important since the support might be dropped one day. I'll check for the other flash failures.

`examples/micropython` and `examples/suit_update` needs investigation as well. The other failures during tests are related to local network setup or are known issues.

**Comment by aabadie** [comment]
For `examples/micropython`, it's strange. The REPL seems to work well but the automated test fails:
```
>>> print("echo this! " * 4)
print("eccho  this! " * 4)
eccho  this! eccho  this! eccho  this! eccho  this! 
```
The command `print("echo this! " * 4)` became `print("eccho this! " * 4)` (duplicated "c" in echo) and I can't reproduce this manually.

For `examples/suit_update`, it was probably a setup issue because when testing manually, it works like a charm.

**Comment by aabadie** [comment]
> I'll check for the other flash failures.

so `tests/gnrc_tcp` and `tests/periph_timer_short_relative_set` could be flashed and are functional. `tests/nordic_softdevice` can also be flashed but it doesn't work.

**Comment by aabadie** [comment]
I also enabled the DC/DC converter that is available in the DWM1001 module. This should reduce power consumption.

**Comment by benpicco** [comment]
I mean this is just another generic nrf52 board, I don't expect anything on the *application level* to not work / work that doesn't also work / not work on other nrf52 boards.

**Comment by aabadie** [comment]
Thanks for reviewing and merging!

**Pull request merged**